src/Node.ts

```diff
@@ -2527,6 +2527,42 @@ export abstract class Node<Config extends NodeConfig = NodeConfig> {
     this.off('touchstart.konva');
   }
 
+  /**
+   * determine if node (at least partially) is currently in user-visible area
+   * @method
+   * @param {(Number | Object)} margin optional margin in pixels
+   * @param {Number} margin.x
+   * @param {Number} margin.y
+   * @returns {Boolean}
+   * @name Konva.Node#isOnScreen
+   */
+  isOnScreen(margin?: number | {x: number; y: number;}): boolean {
```

Comment:
will this work for rotated rects ? May be it will be better to rename function to `isClientRectOnScreen` ? Or take in account rotation too

Reply:
I have no idea really... It fully relies on `getClientRect()` - if it accounts for rotation then it will work, if not then it won't.
The method's name you've suggested is somewhat cumbersome for my liking, but I agree that it describes things a little more precisely. If it's between renaming the method and accounting for rotation, then I definitely for the first option. Not just because I don't want to do it, I genuinely think that's enough for the intended purpose.
Please let me know if you're ok with that and I will rename the method.

Reply:
@VahanG `getClientRect()` always returns non-rotated rectagle. But it takes rotation into account inside itself.

To me `isClientRectOnScreen` is much better. It is easy to confuse `isOnScreen()` with `isVisible()`. Also, I think we should remove `isVisible()` check from the method. So we can use the method to make the node visible.

Reply:
Ok, it's done.
`isOnScreen()` -> `isClientRectOnScreen()`
<s>`if(!this.isVisible()) return false;`</s>